### ISSUE TITLE
continually fails with "kotlin.IllegalStateException: No successful result"

### ISSUE THREAD

**Issue by benkuly** [kotest/kotest#3780]
**Which version of Kotest are you using**
5.8.0

The following code fails, because `something()` and therefore `shouldBe` returns `null` and there is [this new elvis operator](https://github.com/kotest/kotest/blob/master/kotest-assertions/kotest-assertions-core/src/commonMain/kotlin/io/kotest/assertions/nondeterministic/continually.kt#L56)
```
 continually(500.milliseconds) {
    something() shouldBe null
}
```

**Comment by kshired** [comment]
@benkuly Can you give me more specific example codes?

I tried to reproduce the bug, but it worked so well I couldn't reproduce it.

```kotlin
class ContinuallyTest : FunSpec() {
   private fun something() : String? = null

   init {
      test("it works?") {
         continually(500.milliseconds) {
            something() shouldBe null
         }
      }
}

```

**Comment by benkuly** [comment]
@kshired 
changing it to the following fails:
```kotlin
class ContinuallyTest : FunSpec() {
   private fun something() : String? = null

   init {
      test("it works?") {
         val test = continually(500.milliseconds) { // <- this is new
            something() shouldBe null
         }
      }
}
```
I don't know why, but in some of our test cases it also fails without the `val test =`.

**Comment by kshired** [comment]
@benkuly Thanks! It will be fixed by #3786
